src/openms/include/OpenMS/METADATA/IdentificationData.h

```diff
@@ -0,0 +1,320 @@
+      Search query, e.g. spectrum or feature.
+    */
+    struct DataQuery: public MetaInfoInterface
+    {
+      InputFileKey input_file_key; // reference into "input_files"
+
+      // spectrum or feature ID (from the file reference by "input_file_key"):
+      String data_id;
+
+      double rt, mz; // position
+
+      // @TODO: ignore RT and m/z for comparisons?
+      bool operator<(const DataQuery& other) const
+        {
+          return std::tie(input_file_key, data_id, rt, mz) <
+            std::tie(other.input_file_key, other.data_id, other.rt, other.mz);
+        }
+
+      // @TODO: ignore RT and m/z for comparisons?
+      bool operator==(const DataQuery& other) const
+        {
+          return std::tie(input_file_key, data_id, rt, mz) ==
+            std::tie(other.input_file_key, other.data_id, other.rt, other.mz);
+        }
+    };
+
+    typedef UInt64 DataQueryKey;
+    typedef boost::bimap<DataQueryKey, DataQuery> QueryBimap;
+    QueryBimap data_queries;
+
+
+    // Identified molecules - at the moment, peptides or small molecules:
+    typedef UInt64 IdentifiedMoleculeKey;
+    typedef boost::bimap<IdentifiedMoleculeKey, AASequence> PeptideBimap;
+    typedef boost::bimap<IdentifiedMoleculeKey, String> CompoundBimap;
+    PeptideBimap identified_peptides;
+    CompoundBimap identified_compounds;
+
+    enum MoleculeType
+    {
+      MT_PROTEIN,
+      MT_COMPOUND,
+      // MT_RNA,
+      SIZE_OF_MOLECULETYPES
+    };
+
+    /*!
+      Meta data for an identified molecule.
+    */
+    struct IdentifiedMetaData: public MetaInfoInterface
+    {
+      enum MoleculeType molecule_type;
+
+      std::unordered_map<ScoreTypeKey, double> scores;
+
+      std::vector<ProcessingStepKey> processing_steps;
+    };
+
+    std::unordered_map<IdentifiedMoleculeKey,
+                       IdentifiedMetaData> identified_meta_data;
+
+
+    // Evidence linking identified molecules to parent molecules (e.g. peptides
+    // to proteins):
+    // @TODO: rename "PeptideEvidence" to "MoleculeEvidence" (incl. members)
+    typedef std::unordered_map<IdentifiedMoleculeKey,
+                               std::vector<PeptideEvidence>> EvidenceMap;
+    EvidenceMap parent_evidence;
+
+
+    /*!
+      Meta data specific to an identified compound (small molecule).
+    */
+    struct CompoundMetaData: public MetaInfoInterface
+    {
+      EmpiricalFormula formula;
+
+      String smile;
+
+      String inchi;
+    };
+
+    std::unordered_map<IdentifiedMoleculeKey,
+                       CompoundMetaData> compound_meta_data;
+
+
+    /*!
+      Meta data for a search hit (e.g. peptide-spectrum match).
+    */
+    struct MatchMetaData: public MetaInfoInterface
+    {
+      Int charge;
+
+      std::unordered_map<ScoreTypeKey, double> scores;
+
+      // is it useful to store this, as different processing steps/score types
+      // may give different rankings?
+      Size rank; // rank among matches for the same spectrum/feature
```

Comment:
I would love to get rid of it but this might be some work. a rank could always be modelled as score type